tests/components/aemet/test_config_flow.py

```diff
@@ -0,0 +1,148 @@
+"""Define tests for the AEMET OpenData config flow."""
+
+from unittest.mock import patch
+
+import requests_mock
+
+from homeassistant import data_entry_flow
+from homeassistant.components.aemet.const import (
+    DEFAULT_FORECAST_MODE,
+    DOMAIN,
+    FORECAST_MODE_DAILY,
+    FORECAST_MODE_HOURLY,
+)
+from homeassistant.config_entries import (
+    ENTRY_STATE_LOADED,
+    ENTRY_STATE_NOT_LOADED,
+    SOURCE_USER,
+)
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+import homeassistant.util.dt as dt_util
+
+from .util import aemet_requests_mock
+
+from tests.common import MockConfigEntry
+
+CONFIG = {
+    CONF_NAME: "aemet",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 40.30403754,
+    CONF_LONGITUDE: -3.72935236,
+    CONF_MODE: DEFAULT_FORECAST_MODE,
+}
+
+
+async def test_form(hass):
+    """Test that the form is served with valid input."""
+
+    now = dt_util.parse_datetime("2021-01-09 12:00:00+00:00")
+    with patch("homeassistant.util.dt.now", return_value=now), patch(
+        "homeassistant.util.dt.utcnow", return_value=now
+    ), requests_mock.mock() as _m:
+        aemet_requests_mock(_m)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == SOURCE_USER
+        assert result["errors"] == {}
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        )
+
+        await hass.async_block_till_done()
+
+        conf_entries = hass.config_entries.async_entries(DOMAIN)
+        entry = conf_entries[0]
+        assert entry.state == ENTRY_STATE_LOADED
+
+        await hass.config_entries.async_unload(conf_entries[0].entry_id)
```

Comment:
Testing setting up and unloading the config entry should be part of tests for the init module, ie not part of the config flow tests.

Reply:
As above, this is inherited from OpenWeatherMap: https://github.com/home-assistant/core/blob/fd177441b3cfddb22b893b26e91862307bbcc437/tests/components/openweathermap/test_config_flow.py#L62

Reply:
@MartinHjelmare honestly, I have no idea on how you want me to do this.
Can you give me an example of an integration that does it the way you want it or any other kind of help?

Reply:
Here's an example from nws:
https://github.com/home-assistant/core/blob/479ff92acbc9ec4db09b7a300d2beb67f00ab746/tests/components/nws/test_init.py#L10-L37

It's good except for the hass.data checks that we should avoid.